src/otx/engine/engine.py

```diff
@@ -778,6 +779,80 @@ def explain(
         model.explain_mode = False
         return predict_result
 
+    def benchmark(
+        self,
+        checkpoint: PathLike | None = None,
+        batch_size: int = 1,
+        n_iters: int = 10,
+        extended_stats: bool = False,
+    ) -> dict[str, str]:
+        checkpoint = checkpoint if checkpoint is not None else self.checkpoint
+
+        if checkpoint is not None:
+            is_ir_ckpt = Path(checkpoint).suffix in [".xml"]
+            if is_ir_ckpt and not isinstance(self.model, OVModel):
+                # create OVModel
+                self.model = self._auto_configurator.get_ov_model(
+                    model_name=str(checkpoint),
+                    label_info=self.datamodule.label_info,
+                )
+
+            if not is_ir_ckpt:
+                model_cls = self.model.__class__
+                self.model = model_cls.load_from_checkpoint(
+                    checkpoint_path=checkpoint,
+                    map_location="cpu",
+                    **self.model.hparams,
+                )
+        elif isinstance(self.model, OVModel):
+            msg = "To run benchmark on OV model, checkpoint must be specified."
+            raise RuntimeError(msg)
+
+        self.model.eval()
+
+        def dummy_infer(model: OTXModel, batch_size: int = 1, extra_stats: bool = False) -> dict[str, Any]:
+            input_batch = model.get_dummy_input(batch_size)
+            stats = {}
+            start = time.time()
+            if extra_stats:
+                from torch.utils.flop_counter import get_suffix_str, convert_num_with_suffix
+
+                model_fwd = lambda: model.forward(input_batch)
+                depth = 3 if extended_stats else 0
+                fwd_flops = measure_flops(model.model, model_fwd, print_stats_depth=depth)
+                stats["flops"] = convert_num_with_suffix(fwd_flops, get_suffix_str(fwd_flops*10**4))
+
+                params_num = sum(p.numel() for p in self.model.parameters() if p.requires_grad)
+                stats["params"] = convert_num_with_suffix(params_num, get_suffix_str(params_num*100))
+            else:
+                model.forward(input_batch)
+            end = time.time()
+            stats["elapsed"] = end - start
+
+            return stats
+
+        warmup_iters = max(1, int(n_iters / 10))
+        for _ in range(warmup_iters):
+            dummy_infer(self.model, batch_size)
+
+        total_time = 0
+        for _ in range(n_iters):
+            inference_stats = dummy_infer(self.model, batch_size)
+            total_time += inference_stats["elapsed"]
+        total_time /= (n_iters * batch_size)
+
+        final_stats = {"latency": f"{total_time:.3f} s", "troughput": f"{(1 / total_time):.3f} FPS"}
```

Comment:
As far as I know, latency maens how long do I need to wait for getting result from the model. But current latency is measured as `total_time / (n_iters * batch_size)`. I think actual latency can be different because model returns a result after processing whole batch. How do you think about it?